csharp/src/Google.Protobuf.Test/CodedInputStreamAsyncTest.cs

```diff
@@ -0,0 +1,502 @@
+#region Copyright notice and license
```

Comment:
Most tests are similar with the sync version. We'd better to make the them share the test code. So if
we want to change some code in future, the share test code can remind us to change both sync and async.

Reply:
As with all streams, the .net standard is to have the legacy blocking methods, and then have the same signature but async keyword and their named suffixed with "Async". There's no need for all this switching.